modules/core/02-client/keeper/keeper.go

```diff
@@ -359,15 +359,16 @@ func (k Keeper) IterateClients(ctx sdk.Context, cb func(clientID string, cs expo
 
 	defer iterator.Close()
 	for ; iterator.Valid(); iterator.Next() {
-		keySplit := strings.Split(string(iterator.Key()), "/")
-		if keySplit[len(keySplit)-1] != host.KeyClientState {
+		path := string(iterator.Key())
+		if !strings.Contains(path, host.KeyClientState) {
+			// skip non client state keys
 			continue
```

Comment:
will this ever be hit?

Reply:
Yes for consensus state keys and metadata keys

Reply:
Since client state keys are at the end of the key, you cannot prefix iterate on them